lib/db/mysql/index.js

```diff
@@ -299,7 +300,7 @@ MysqlStore.prototype = {
     if (!client.id) {
       return P.reject(new Error('Update client needs an id'));
     }
-    var secret = client.hashedSecret || client.secret || null;
+    var secret = client.hashedSecret;
```

Comment:
I didn't really feel a new here, as they're just local variables, and the intent is clear.